src/js/workshop-comments.js

```diff
@@ -0,0 +1,132 @@
+			// Show successful comment submission message.
+			applyVisibility(document.querySelector(".submit-success-message"), true , "block");
+			// Clear Text Fields.
+			commentForm.reset();
+		} else {
+			// Show failed comment submission message.
+			submitFailureMessage.setAttribute("style", "display: block;");
+		}
+		disableFields(false);
+	});
+
+	// Show failure message in case of error.
+	XHR.addEventListener( "error", function() {
+		// Show failed comment submission message.
+		submitFailureMessage.setAttribute("style", "display: block;");
+		disableFields(false);
+	} );
+
+	// Set up our request.
+	XHR.open( "POST", window.location.origin + "/api/comments" );
+
+	// The data sent is what the user provided in the form.
+	const jsonFormData = JSON.stringify(Object.fromEntries(commentFormData));
+	XHR.setRequestHeader("Content-Type", "application/json");
+	XHR.send(jsonFormData);
+});
+
+/*
+ * Hide/show required and success messages.
+ * @param {DOMElement} domSelector - The DOM Element to parse.
+ * @param {Boolean} toShow - value equals true to show and equals false to hide.
+ * @param {String}[Optional] displayType - CSS display value to apply. Only applicable when toShow is set to true.
+ */
+function applyVisibility(domSelector, toShow, displayType) {
+	displayType = toShow ? displayType : "none";
+	domSelector.setAttribute("style", `display:  ${displayType};`);
+}
+
+/*
+ * Display the new comment on the current page.
+ * @param {DOMElement} nameField - DOM element of name field in comment form.
+ * @param {DOMElement} commentField - DOM element of comment field in comment form.
+ */
+function showSubmittedComment(nameField, commentField) {
+	// Add comment to the top of the comments section of workshop page.
+	const newCommentDiv = document.createElement("div");
+	newCommentDiv.innerHTML = "<article class='comment submitted-comment'><p><span class='comment-name'></span> | <span class='comment-date'></span></p><p class='comment-text'></p></article>";
+	const commentsDiv = document.getElementsByClassName("comments")[0];
+	commentsDiv.prepend(newCommentDiv.firstChild);
+
+	const submittedCommentDiv = commentsDiv.firstChild;
+	const submittedDate = submittedCommentDiv.querySelector(".comment-date");
+	const submittedName = submittedCommentDiv.querySelector(".comment-name");
+	const submittedComment = submittedCommentDiv.querySelector(".comment-text");
+
+	const dateObject = new Date();
+
+	// Add values of comment.
+	submittedDate.innerText = convertDate(dateObject);
+	submittedName.innerText = nameField.value.trim();
+	submittedComment.innerText = commentField.value.trim();
+}
+
+/*
+ * Disable the "post a comment" button and text fields when the post request is in the process of sending.
+ * @param {Boolean} sendingStatus - Boolean value indicating whether or not post request is in the send state.
+ */
+function disableFields(sendingStatus) {
+	const backgroundColor = sendingStatus ? "#f3f3f3" : "#fff";
```

Comment:
Add a code comment about changing background color only applies when not applying UIO themes. When an UIO theme is selected, the background color will remain the same of what's set by the theme.